src/lib/core/a11y/focus-trap.ts

```diff
@@ -114,4 +158,55 @@ export class FocusTrap {
 
     return null;
   }
+
+  private _createAnchor(): HTMLElement {
+    let anchor = document.createElement('div');
+    anchor.tabIndex = 0;
+    anchor.classList.add('cdk-visually-hidden');
+    anchor.classList.add('cdk-focus-trap-anchor');
+    return anchor;
+  }
+}
+
+
+/**
+ * Service that allows easy instantiation of focus traps.
+ */
+@Injectable()
+export class FocusTrapService {
```

Comment:
Ok that's fine with me, but Jermey would be upset with me if I let you name it Service :p how about `FocusTrapFactory` and change `attach` to `create` (since the attachment part is kind of optional anyways)